### PR TITLE
updating the haproxy configuration and adding packaging scripts

### PR DESCRIPTION
**Pull request by Gentleelephant**

### CODE REVIEW

File: charts/vector/values-agent.yaml [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+  # secrets.generic -- Each Key/Value will be added to the Secret's data key, each value should be raw and NOT base64
+  # encoded. Any secrets can be provided here. It's commonly used for credentials and other access related values.
+  # **NOTE: Don't commit unencrypted secrets to git!**
+  generic: {}
+    # my_variable: "my-secret-value"
+    # datadog_api_key: "api-key"
+    # awsAccessKeyId: "access-key"
+    # awsSecretAccessKey: "secret-access-key"
+
+autoscaling:
```
Comment (benjaminhuo) [inline charts/vector/values-agent.yaml:60]: autoscaling is not necessary for agent https://github.com/WhizardTelemetry/vector-helm-charts/pull/2/files#diff-033177f7a15cb8a53378f2a323022c089d4e39f4601c9469a6b25457436c4b16R60-R86
```shell
  # autoscaling.enabled -- Create a [HorizontalPodAutoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/)
  # for Vector. Valid for the "Aggregator" and "Stateless-Aggregator" roles.
````

File: charts/vector/values-agent.yaml [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+  additionalLabels: {}
+  # podMonitor.honorLabels -- If true, honor_labels is set to true in the [scrape config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).
+  honorLabels: false
+  # podMonitor.honorTimestamps -- If true, honor_timestamps is set to true in the [scrape config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).
+  honorTimestamps: true
+
+# Log level for Vector.
+logLevel: "info"
+
+# Optional built-in HAProxy load balancer.
```
Comment (benjaminhuo) [inline charts/vector/values-agent.yaml:456]: haproxy is not necessary for agent: https://github.com/WhizardTelemetry/vector-helm-charts/pull/2/files#diff-033177f7a15cb8a53378f2a323022c089d4e39f4601c9469a6b25457436c4b16R456-R623

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

File: charts/vector/values-agent.yaml (hunk 2 above)
Comment (Gentleelephant) [reply charts/vector/values-agent.yaml:456]: agent is false

File: charts/vector/values-agent.yaml (hunk 1 above)
Comment (Gentleelephant) [reply charts/vector/values-agent.yaml:60]: The default configurations copied from values are all false.

File: charts/vector/values-agent.yaml (hunk 2 above)
Comment (benjaminhuo) [reply charts/vector/values-agent.yaml:456]: we can remove these config because they'll never be applied for agent

Comment (Gentleelephant) [reply charts/vector/values-agent.yaml:456]: It is kept for comparison with upstream modifications.

Comment (Gentleelephant) [reply charts/vector/values-agent.yaml:456]: image

**Pull request merged**